### ISSUE TITLE
Unprotecting a worksheet with a password specified when the sheet was not protected with a password results in any future programmatic protect calls to not apply any new specified passwords

### ISSUE THREAD

**Issue by Sheikaa3** [OfficeDev/office-js#762]
## Expected Behavior
I'm currently working on an Excel add-in and ran into the following issue: 

After the user protects a worksheet manually, I unprotect the sheet, do some add-in behaviour on the sheet and then protect it back. In the workflow, when a user protects a sheet without a password, I apply a default password during the re-protect process. This should result in the sheet being protected with the latest specified password, and the user should see the password prompt appear if they try to unprotect the sheet manually following this. 

## Current Behavior
However, I'm seeing that on the latest version of Excel (1909 on the Monthly Build Channel) the password isn't being applied. The user is able to manually unprotect the sheet and is not prompted to enter a password, despite the password being applied.

I was able to pinpoint the problem's cause to the Unprotect call having a password as a parameter when the sheet was protected without a password. Testing that case, I found that the bug never occurred when the sheet was unprotected without a password after having been protected without a password. It appears that the root cause is some sort of error where unprotecting a sheet that was protected without a password doesn't result in the sheet's password being changed (in the sheet's .xml maybe?) and when the sheet is protected again the old password is retained instead of being overridden with the new password. This behaviour doesn't happen in version 1908 of Excel.

## Steps to Reproduce, or Live Example
The following code snippet is an example of what I'm doing in the add-in;

```
//sheet is manually protected by the user without a password prior to this code running
_worksheet._Unprotect(pw);

//internal logic goes here

_worksheet._Protect(pw)
```
I was able to replicate the issue purely in vba with the following;
```
Sub Macro1()
    ActiveSheet.Protect
    ActiveSheet.Unprotect ("password")
    ActiveSheet.Protect ("password")
End Sub
```

## Your Environment

* Platform [PC desktop, Mac, iOS, Office Online]: PC
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: 1909
* Operating System: Windows 10

**Comment by catchingyu** [comment]
Thanks a lot for reaching us. 

One clarification on the reproduce steps:
1. Unprotected the file with the old pwd
2. Protected the file again with the new pwd

Expected : 
We expect the new pwd should work

Current Behavior : 
The file just can be opened successfully without the input of the pwd.

**Comment by Sheikaa3** [comment]
@catchingyu Hello, I just wanted to verify that you were also able to reproduce the issue with the following steps:

1. Protect file without using a password
2. Unprotect with a password provided
3. Protect with a new password

Expected (and the behaviour in 1908):
The new password should work

Current Behaviour:
The file can be unprotected successfully without the input of a password

**Comment by Sheikaa3** [comment]
We're curious about the process as well and what your timeline looks like, if you can share any information regarding that. We have to choose whether to push out a fix that would compensate for this behaviour so any further info would be appreciated. Thank you!

**Comment by catchingyu** [comment]
Team is actively working on this issue. bug 3723631 is created to track this issue. 

However, with your reproduce steps, are you doing that all through the API, right? The last step 'the file can be unprotected successfully without the input of a pwd', are you saying unprotect the file using JS API or just using the Excel App to open it?

**Comment by MandytMSFT** [comment]
@Sheikaa3  i can not repro your issue, but i am on 1911. What kind of protection you are using? i mean to something like "Insert rows"/"Insert columns".

**Comment by Sheikaa3** [comment]
@catchingyu All of the steps are done through the Excel Interop API (found here: https://docs.microsoft.com/en-us/dotnet/api/microsoft.office.interop.excel?view=excel-pia) or through VBA as described. In the case of the unprotect, that was tested both with VBA and the interop calls and the problem was reproducible.

@MandytMSFT In both the interop and VBA I was performing the protect with no other parameters besides the password. The rest of the protect should be happening according to the default params (found here: https://docs.microsoft.com/en-us/office/vba/api/excel.worksheet.protect). Just to clarify, my Excel has updated to version 1910 and I am still able to replicate the issue by doing the following:

1. Create a new workbook with one worksheet
2. Manually protect the worksheet without a password
3. Perform the following VBA macro:
4. `Sub Macro1()
     ActiveSheet.Unprotect ("test")
     ActiveSheet.Protect ("test")
 End Sub`
9. Manually unprotect the sheet

Expected behaviour: The unprotect will request a password, which will be "test"
Actual behaviour: The unprotect succeeds without requiring a password to be entered

**Comment by MandytMSFT** [comment]
@Sheikaa3 Got you. I can repro by VBA. We are actively working on this.

**Comment by donlvMSFT** [comment]
This issue already been fixed from the bug status, and released after Office version 16.0.12509.34851, please help verify. I'd like to close this thread.
